vl53v1x/vl53l1x.go

```diff
@@ -0,0 +1,541 @@
+// configured.
+//
+// This function only creates the Device object, it does not touch the device.
+func New(bus machine.I2C) Device {
+	return Device{
+		bus:     bus,
+		mode:    LONG,
+		timeout: 500,
+	}
+}
+
+// Connected returns whether a VL53L1X has been found.
+// It does a "who am I" request and checks the response.
+func (d *Device) Connected() bool {
+	return d.readReg16Bit(WHO_AM_I) == CHIP_ID
+}
+
+// Configure sets up the device for communication
+func (d *Device) Configure(use2v8Mode bool) bool {
+	if !d.Connected() {
+		return false
+	}
+	d.writeReg(SOFT_RESET, 0x00)
+	time.Sleep(100 * time.Microsecond)
+	d.writeReg(SOFT_RESET, 0x01)
+	time.Sleep(1 * time.Millisecond)
+
+	start := time.Now()
+	for (d.readReg(FIRMWARE_SYSTEM_STATUS) & 0x01) == 0 {
+		elapsed := time.Since(start)
+		if d.timeout > 0 && uint32(elapsed.Seconds()*1000) > d.timeout {
+			return false
+		}
+	}
+
+	if use2v8Mode {
+		d.writeReg(PAD_I2C_HV_EXTSUP_CONFIG, d.readReg(PAD_I2C_HV_EXTSUP_CONFIG)|0x01)
+	}
+
+	d.fastOscillatorFreq = d.readReg16Bit(OSC_MEASURED_FAST_OSC_FREQUENCY)
+	d.oscillatorOffset = d.readReg16Bit(RESULT_OSC_CALIBRATE_VAL)
+
+	// static config
+	d.writeReg16Bit(DSS_CONFIG_TARGET_TOTAL_RATE_MCPS, TARGETRATE)
+	d.writeReg(GPIO_TIO_HV_STATUS, 0x02)
+	d.writeReg(SIGMA_ESTIMATOR_EFFECTIVE_PULSE_WIDTH_NS, 8)
+	d.writeReg(SIGMA_ESTIMATOR_EFFECTIVE_AMBIENT_WIDTH_NS, 16)
+	d.writeReg(ALGO_CROSSTALK_COMPENSATION_VALID_HEIGHT_MM, 0xFF)
+	d.writeReg(ALGO_RANGE_MIN_CLIP, 0)
+	d.writeReg(ALGO_CONSISTENCY_CHECK_TOLERANCE, 2)
+
+	// general config
+	d.writeReg16Bit(SYSTEM_THRESH_RATE_HIGH, 0x0000)
+	d.writeReg16Bit(SYSTEM_THRESH_RATE_LOW, 0x0000)
+	d.writeReg(DSS_CONFIG_APERTURE_ATTENUATION, 0x38)
+
+	// timing config
+	d.writeReg16Bit(RANGE_CONFIG_SIGMA_THRESH, 360)
+	d.writeReg16Bit(RANGE_CONFIG_MIN_COUNT_RATE_RTN_LIMIT_MCPS, 192)
+
+	// dynamic config
+	d.writeReg(SYSTEM_GROUPED_PARAMETER_HOLD_0, 0x01)
+	d.writeReg(SYSTEM_GROUPED_PARAMETER_HOLD_1, 0x01)
+	d.writeReg(SD_CONFIG_QUANTIFIER, 2)
+
+	d.writeReg(SYSTEM_GROUPED_PARAMETER_HOLD, 0x00)
+	d.writeReg(SYSTEM_SEED_CONFIG, 1)
+
+	// Low power auto mode
+	d.writeReg(SYSTEM_SEQUENCE_CONFIG, 0x8B) // VHV, PHASECAL, DSS1, RANGE
+	d.writeReg16Bit(DSS_CONFIG_MANUAL_EFFECTIVE_SPADS_SELECT, 200<<8)
+	d.writeReg(DSS_CONFIG_ROI_MODE_CONTROL, 2) // REQUESTED_EFFFECTIVE_SPADS
+
+	d.SetDistanceMode(d.mode)
+	d.SetMeasurementTimingBudget(50000)
+
+	d.writeReg16Bit(ALGO_PART_TO_PART_RANGE_OFFSET_MM, d.readReg16Bit(MM_CONFIG_OUTER_OFFSET_MM)*4)
+
+	return true
+}
+
+// SetTimetou configures the timeout
+func (d *Device) SetTimeout(timeout uint32) {
+	d.timeout = timeout
+}
+
+// SetDistanceMode sets the mode for calculating the distance.
+// Distance mode vs. max. distance
+// SHORT: 136cm (dark) - 135cm (strong ambient light)
+// MEDIUM: 290cm (dark) - 76cm (strong ambient light)
+// LONG: 360cm (dark) - 73cm (strong ambient light)
+func (d *Device) SetDistanceMode(mode DistanceMode) bool {
```

Comment:
Can you add a description of the return value to the comment here?
In this case, it seems to be that it returns whether the `mode` is a valid DistanceMode.